test/stdlib/Mirror.swift

```diff
@@ -782,4 +782,20 @@ mirrors.test("String.init") {
   expectEqual("42", String(reflecting: 42))
   expectEqual("\"42\"", String(reflecting: "42"))
 }
+
+struct a<b>  {
+    enum c{}
+}
+class d  {}
+struct e<f> {
+    var constraints: [Int: a<f>.c] = [:]
+}
```

Comment:
Another silly case to test is a type nested inside an extension of a generic type that makes one of its generic parameters concrete, or more generally, any constrained extension

Reply:
Yeah, having a couple levels of generic/non-generic nesting interleaved is great for sussing out bugs in this code. `test/Runtime/demangleToMetadata.swift` has some examples.